### PR TITLE
Austin/codable

### PR DESCRIPTION
**Pull request by AAAstorga**
#51 
#46 
#45 
#43 
#35 
Pretty big PR. I removed Realm! updated everything to Coddle (so beautiful). I added a Video Cell +  Photo gallery logic. All done in house. Yeet

### CODE REVIEW

File: Cornell Sun/Cornell Sun/Cells/CaptionCell.swift [hunk 1]
```diff
@@ -42,7 +44,12 @@ final class CaptionCell: UICollectionViewCell {
     }
 
     func updateCaption(index: Int) {
-        captionLabel.text = post?.photoGalleryObjects[index].caption
+        captionLabel.text = post?.postAttachments[index].caption ?? ""
+        captionLabel.snp.remakeConstraints { make in
+            make.top.equalToSuperview().inset(10)
+            make.left.equalToSuperview().inset(18)
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Cells/CaptionCell.swift]: make these insets constants

File: Cornell Sun/Cornell Sun/Cells/VideoCell.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    var post: PostObject? {
+        didSet {
+            setupVideo()
+        }
+    }
+
+    let videoWebView: UIWebView = {
+        let webView = UIWebView()
+        webView.allowsInlineMediaPlayback = true
+        webView.backgroundColor = .orange
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Cells/VideoCell.swift]: why is the background orange? lol

File: Cornell Sun/Cornell Sun/Cells/VideoCell.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setupViews() {
+        addSubview(videoWebView)
+        videoWebView.snp.makeConstraints { (make) in
+            make.width.equalToSuperview()
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Cells/VideoCell.swift]: chain `width.height` together

File: Cornell Sun/Cornell Sun/Cells/VideoCell.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        setupViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setupViews() {
+        addSubview(videoWebView)
+        videoWebView.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Cells/VideoCell.swift]: remove parentheses for `(make)`

File: Cornell Sun/Cornell Sun/Cells/PhotoGalleryCell.swift [hunk 5]
```diff
@@ -11,75 +11,56 @@ import IGListKit
...
-
     func setupViews() {
-        let gestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didTapPhotos))
-        slideShow.addGestureRecognizer(gestureRecognizer)
+        photoGallery = PhotoGallery(attachments: post!.postAttachments, height: self.bounds.height, width: self.bounds.width)
 
-        addSubview(slideShow)
-        slideShow.snp.makeConstraints { (make) in
+        addSubview(photoGallery)
+        photoGallery.snp.makeConstraints { (make) in
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Cells/PhotoGalleryCell.swift]: remove parentheses around `(make)`

File: Cornell Sun/Cornell Sun/Networking/NetworkingHelpers.swift [hunk 6]
```diff
@@ -19,8 +18,23 @@ enum APIErrors: Error {
 typealias PostObjectCompletionBlock = (_ posts: [PostObject], _ error: APIErrors?) -> Void
 typealias TrendingCompletionBlock = (_ trending: [String], _ error: APIErrors?) -> Void
 typealias CommentsCompletionBlock = (_ comments: [CommentObject], _ error: APIErrors?) -> Void
-let savedPosts: Results<PostObject> = RealmManager.instance.get()
-let savedPostIds: [Int] = Array(RealmManager.instance.get()).map({$0.id})
+
+let savedPostIds: [Int] = {
+    guard let posts = PostOffice.instance.get() else { return [] }
+    return posts.map({$0.id})
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Networking/NetworkingHelpers.swift]: remove parentheses around `{ $0.id }`

File: Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+        if !didSetupConstraints {
+
+            collectionView.snp.makeConstraints { make in
+                make.edges.equalToSuperview()
+            }
+
+            pageControl.snp.makeConstraints { make in
+                make.height.equalTo(pageControl.intrinsicContentSize.height)
+                make.bottom.equalToSuperview().inset(4)
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift]: make this a constant

File: Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        return attachments.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "image-cell", for: indexPath) as? ImageCollectionCell else { return UICollectionViewCell() }
+
+        cell.imageView.kf.setImage(with: attachments[indexPath.item].url, progressBlock: { receivedSize, totalSize in
+            let percentage = (CGFloat(receivedSize) / CGFloat(totalSize)) * 100.0
+            cell.updatePercentage(percentage: percentage)
+        })
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift]: does the `})` have an extra parentheses?

File: Cornell Sun/Cornell Sun/Post Office/PostOffice.swift [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    override init() {
+        self.packages = []
+        super.init()
+        self.packages = get() ?? []
+    }
+
+    func get() -> [PostObject]? {
+        guard let packageData = UserDefaults.standard.data(forKey: packagesKey) else { return nil }
+            let packages = try? decoder.decode([PostObject].self, from: packageData)
```
Comment (mindylou) [inline Cornell Sun/Cornell Sun/Post Office/PostOffice.swift]: remove indent

File: Cornell Sun/Cornell Sun/Cells/VideoCell.swift (hunk 2 above)
Comment (AAAstorga) [reply Cornell Sun/Cornell Sun/Cells/VideoCell.swift]: oh lol left over from testing my b

File: Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift (hunk 8 above)
Comment (AAAstorga) [reply Cornell Sun/Cornell Sun/PhotoGallery/PhotoGallery.swift]: no, it is part of `setImage()`

**Pull request merged**